src/transformers/modeling_utils.py

```diff
@@ -2876,9 +2882,16 @@ def _fix_key(key):
         unexpected_keys = list(set(loaded_keys) - set(expected_keys))
 
         # Some tensors maybe have been already filled by another key (tied weights).
-        existing_ptrs = {model_state_dict[k].data_ptr() for k in loaded_keys if k in model_state_dict}
+        existing_ptrs = {
+            (model_state_dict[k].data_ptr(), model_state_dict[k].device)
+            for k in loaded_keys
+            if k in model_state_dict and model_state_dict[k].device != torch.device("meta")
+        }
```

Comment:
This is for the ViT `fp16` issue on the slow test.

If the state_dict is on device_meta, then ALL `data_ptr()` are `0` so this would make all keys to be ignored from `missing_,keys` which I'm guessing accelerate relies on to do the gpu attribution.

Also found an issue on multi-gpu where the autoload would clash in this particular case.
@ydshieh wasn't able to reproduce on the ci-runner, but it's deterministic and can be triggered on 4.27.4 with accelerate==0.18.0

Reply:
This will ignore the whole state dict as soon as `device_map="auto"` or `low_cpu_mem_usage=True`.

I'm not too sure why we are not using Accelerate to detect the tied weights at this stage.